rust/lance-index/src/vector/v3/shuffler.rs

```diff
@@ -297,3 +299,33 @@ impl ShuffleReader for IvfShufflerReader {
         Ok(self.partition_sizes[partition_id])
     }
 }
+
+pub struct SinglePartitionReader {
+    data: sync::Mutex<Option<Box<dyn RecordBatchStream + Unpin + 'static>>>,
```

Comment:
small nit: `sync::Mutex` read like it might be a non-threadsafe mutex. Let's use the full name `tokio::sync::Mutex` or just `Mutex`